dashboard/cookbook/jstranslation.md

```diff
@@ -0,0 +1,229 @@
+
+##### URL
+
+Variables can contain either *the project URL* or some other important location (such as that of a *linked project*) that you would prefer to have **remapped** over the proxy. Don't give in to the temptation to localize URLs in JS as plain text! Instead, use the `url` postfix to map them:
+
+```
+"%"."exampleUrl" url
+```
+
+##### HTML
+
+`exampleHtmlString` demonstrates the fact that JS variables frequently hold markup (for better or worse). The `html` postfix lets you process these strings as HTML.
+
+```
+"%"."exampleHtmlString" html[@process]
+```
+
+`[@process]` is optional. By adding it, you instruct the proxy to apply the translation-invariable regular expressions currently set on the project.
+
+##### Nested Javascript
+
+Although JS paths are mostly specified in a single line, the `javascript` postfix bends this rule. It tells the proxy to apply the rule in the *next line* to the value of the postfixed JSON path. One level of nesting is supported. It is rarely needed, but invaluable when it is called for.
+
+Plain text:
+
+```
+"%"."exampleObject"."exampleNestedJS" javascript
+"%"."exampleObject"."exampleNestedJS"."nestedVar"."nestedKey"
+```
+
+HTML:
+
+```
+"%"."exampleObject"."exampleNestedHTMLinJS" javascript
+"%"."exampleObject"."exampleNestedHTMLinJS"."nestedHTML" html
+```
+Note that the JSON Path tester tool is not equipped to display the nested use case.
+
+### Xpaths
+
+Xpaths for XML AJAX responses in a similar way as JS/JSON paths do for their respective content types. For example, `/ajax-response/component[1]/text() html`
+assumes that the first `<component>` node contains translatable HTML markup.
+
+Due to space constraints, we stop at this brief mention and decline to reproduce a full Xpath tutorial in this documentation. We direct the reader's attention to the many tutorial resources available online.
+
+### Limitations
+
+#### Content extraction
+
+A simple content extraction crawl takes care of JS content in the source of an HTML document. But in many cases, such content is requested **as part of a page load** or **via user action**. The same limitation still applies, however: the page is not available to the crawler in a way that would allow for such "interactive" requests to start.
+
+The solution is to extract content **via Preview**. Open the page in Preview mode and go through the required user actions to trigger all necessary events. The proxy will take care of extracting content from the affected dynamic responses (provided that your setup is correct).
+
+Note the influence of **TM Freeze** on this approach: you need to disable it temporarily for Preview-based content extraction to work.
+
+#### Computed Values
+
+JS translation cannot be used with values such as `var string = "You have " + getCartItems().length + " items in your basked"`). In these cases, you either have to forego translation or change the content so that no *computed expression* is present among the concatenated elements.
```

Comment:
@lsoos As far as I remember, we have a solution for the case where the result string is concatenated of smaller strings *where non of them is a computed expression, only a single string*; is it worth to explain it here?